flow/project.py

```diff
@@ -1597,6 +1610,39 @@ def __init__(self, config=None, environment=None, entrypoint=None):
         self._group_to_aggregate_store = _bidict()
         self._register_groups()
 
+    def _check_flow_schema_compatibility(self):
+        """Check whether this project's data schema is compatible with this version.
+
+        Raises
+        ------
+        :class:`~flow.errors.IncompatibleSchemaVersion`
+            If the schema version is incompatible.
+
+        """
+        schema_version = version.parse(SCHEMA_VERSION)
+        config_schema_version = version.parse(
+            flow_config.get_config_value("schema_version", config=self.config)
+        )
+        if config_schema_version > schema_version:
+            # Project config schema version is newer and therefore not supported.
+            raise IncompatibleSchemaVersion(
+                "The signac-flow schema version used by this project is '{}', but flow {} "
+                "only supports up to schema version '{}'. Try updating flow.".format(
```

Comment:
Make this error message consistent with the one in `flow/migration/__init__.py`. The terms `signac-flow` and `flow` are interchanged (and possibly other differences).